main.cpp

```diff
@@ -8,6 +8,9 @@ using namespace BinaryNinja;
 extern "C" {
 BN_DECLARE_CORE_ABI_VERSION;
 
+static Ref<Symbol> sym = nullptr;
```

Comment:
This makes the assumption that there's only one open BV. However, if Binary Ninja has multiple binaries open, there's still only one instance of each plugin loaded, so we'd end up potentially using the wrong symbol address.

I considered doing a map from BV to Symbol address, but I couldn't devise a unique identifier for BVs. I believe one of the examples uses memory address, but that's not _technically_ correct, as you could get cosmically unlucky and allocate a second BV in the same place you just freed one. Am I being unreasonably pedantic? Maybe :)

If you're using this and you're actually hitting a speed bottleneck I'm open to including a cache solution behind an option or something to make your analysis faster.

Reply:
I didn't know that plugins are global to binary ninja. As for a devising a unique identifier for binary views, we could use the fullpath of the backing executable. It's not as unique as I would like, but it's a start that should cover most use cases.

For my uses, I'm not hitting a bottleneck through your plugin. It's already fast enough, but I haven't used a profiler on this yet.

Reply:
After some research I found that FileMetadata has a field titled `GetSessionID()` which produced a UUID that is unique to each binaryview instance. I just pushed a commit that uses that SessionID to pair it with a `Ref<Symbol>` in a static `std::map`.

Reply:
Ah that's cool, that's probably the best we can do with current APIs